packages/docgen/src/compile.js

```diff
@@ -0,0 +1,47 @@
+const ts = require( 'typescript' );
+const { readFileSync } = require( 'fs' );
+
+const { getExportStatements } = require( './get-export-statements' );
+
+/**
+ * Function that takes file path and returns compiled result
+ *
+ * @param {string} filePath the path to the file to be compiled
+ */
+module.exports = function( filePath ) {
+	const options = {
+		allowJs: true,
+		target: ts.ScriptTarget.ES2020,
```

Comment:
It isn't important because this option is for the **transpiled result** of the code. It's not for how TypeScript treats new syntaxes. In other words, even when we set this to `ES3`, TypeScript can compile optional chaining and other ES2015-2020 features. But in this case, the result of the transpilation can be run on the environments that only support ES3 (maybe notorious IE9?).

We don't need transpiled JavaScript in docgen. All we need is the generated AST. So, it doesn't matter. 

But to avoid confusion, I removed target option from `compile.js` and changed `ES2020` to `ESNext` in other files.